### PR TITLE
Extend section on addresses.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -137,15 +137,26 @@ Operators:
...
+
+The idea behind this distinction is that ``address payable`` is an address you can send Ether to,
+while a plain ``address`` cannot be sent Ether.
+
+Type conversions:
 
 Implicit conversions from ``address payable`` to ``address`` are allowed, whereas conversions from ``address`` to ``address payable`` are
 not possible (the only way to perform such a conversion is by using an intermediate conversion to ``uint160``).
+
+Address literals can be implicitly converted to ``address payable``.
```
Comment (axic) [inline docs/types.rst]: Can you link to the address literal section?

File: docs/types.rst [hunk 2]
```diff
@@ -163,25 +174,30 @@ Operators:
 .. note::
     Starting with version 0.5.0 contracts do not derive from the address type, but can still be explicitly converted to
     ``address`` or to ``address payable``, if they have a payable fallback function.
+    The distinction between ``address`` and ``address payable`` was introduced with version 0.5.0.
```
Comment (axic) [inline docs/types.rst]: The previous sentence kind of says this too? Perhaps reword the two sentences into one?

File: docs/types.rst [hunk 3]
```diff
@@ -195,14 +211,23 @@ Send is the low-level counterpart of ``transfer``. If the execution fails, the c
...
-* ``call``, ``callcode``, ``delegatecall`` and ``staticcall``
+* ``call``, ``delegatecall`` and ``staticcall``
 
-Furthermore, to interface with contracts that do not adhere to the ABI,
+In order to interface with contracts that do not adhere to the ABI,
 or to get more direct control over the encoding,
-the function ``call`` is provided which takes a single byte array as input.
+the functions ``call``, ``delegatecall`` and ``staticcall`` are provided.
+They all take a single ``bytes memory`` argument as input and
+returns the success condition (as a ``bool``) and the returned data
```
Comment (axic) [inline docs/types.rst]: returns -> return

File: docs/types.rst (hunk 2 above)
Comment (chriseth) [reply docs/types.rst]: Yep, will merge the sentences.

File: docs/types.rst [hunk 4]
```diff
@@ -230,27 +253,21 @@ Lastly, these modifiers can be combined. Their order does not matter::
...
+All three functions ``call``, ``delegatecall`` and ``staticcall`` are very low-level functions and should only be used as a *last resort* as they break the type-safety of Solidity.
 
 The ``.gas()`` option is available on all three methods, while the ``.value()`` option is not supported for ``delegatecall``.
 
 .. note::
     All contracts can be converted to ``address`` type, so it is possible to query the balance of the
     current contract using ``address(this).balance``.
 
-.. note::
-    The use of ``callcode`` is discouraged and will be removed in the future.
```
Comment (axic) [inline docs/types.rst:252]: Care to mention `callcode` existed but was removed in 0.5.0?

**Review by axic — CHANGES_REQUESTED** [review]

File: docs/types.rst (hunk 2 above)
Comment (chriseth) [reply docs/types.rst]: Hm, didn't really work, but at least I swapped them.

File: docs/types.rst (hunk 4 above)
Comment (axic) [reply docs/types.rst:252]: Never mind, it is there now (the diff was outdated).

Comment (chriseth) [reply docs/types.rst:252]: It is mentioned close to where delegatecall is explained.

**Comment by chriseth** [comment]
Updated.

**Review by axic — APPROVED** [review]

**Pull request merged**